src/index.js

```diff
@@ -1,42 +1,81 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
+
   console.log('Calculating subtotal, yey!');
 
   //... your code goes here
+  let price = product.querySelector (".price span").innerHTML;
+  let quantity = product.querySelector (".quantity input").value;
+  
+  let subTotalPrice = price*quantity;
+  let subTotal = product.querySelector(".subtotal span")
+
+subTotal.innerHTML = subTotalPrice
+
+return subTotalPrice;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+/*   const singleProduct = document.querySelector('.product');
+  updateSubtotal(singleProduct); */
   // end of test
 
   // ITERATION 2
   //... your code goes here
+  const getProducts = document.getElementsByClassName("product")
+  
+  let total = 0;
+  for (let i=0; i<getProducts.length; i++){
+
+    total += updateSubtotal (getProducts[i])
+  }
 
   // ITERATION 3
   //... your code goes here
+
+  document.querySelector("#total-value span").innerHTML = total;
+  return total
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  //O user clicou por último
+  console.log('The target in remove is:', target.parentNode.parentNode);
+ target.parentNode.parentNode.remove()
 }
 
 // ITERATION 5
 
-function createProduct() {
+function createProduct(event) {
   //... your code goes here
-}
 
+  //Add a click event handler to the "Create Product" that will take a function named createProduct as a callback.
+ 
+  //TO BE COMPLETED...
+ 
```

Comment:
Here we need to target the inputs and store their values, target the table, create a new element and try to pass the values from the inputs to the new created element.
Here's a solution you could've used:
```
function createProduct() {
  // Targeting all the create product inputs that the user will fill
  let newProductInputs = document.querySelectorAll(".create-product input");

  // Save the value of the inputs in variables
  let newProductName = newProductInputs[0].value;
  let newProductPrice = newProductInputs[1].value;

  // Saving the newProduct html code in a variable
  let newProduct = `
  <tr class="product">
    <td class="name">
      <span>${newProductName}</span>
    </td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</button>
    </td>
  </tr>`;

  // Targeting the table
  const tableBody = document.querySelector("tbody");

  // Adding the newProduct to the end of the table
  tableBody.insertAdjacentHTML("beforeend", newProduct);

  // Adding the removeProduct function to the remove button
  let removeBtn = document.querySelectorAll(".btn-remove");
  removeBtn.forEach((btn) => btn.addEventListener("click", removeProduct));
}
```